R/functions_netCDF.R

```diff
@@ -230,268 +512,307 @@ create_netCDF_from_raster_with_variables <- function(x, time_bounds,
     }
   }
 
-  if (!has_time_central) {
+  if (!has_T_timeAxis) {
     ncdf4::ncatt_put(nc, varid = 0, attname = "time_label",
       attval = "None")
     ncdf4::ncatt_put(nc, varid = 0, attname = "time_title",
       attval = "No temporal dimensions ... fixed field")
   }
 
-  # add coordinate system attributes
-  prj <- raster::crs(x)
-  if (!is.na(prj)) {
-    ncdf4::ncatt_put(nc, varid = "crs", attname = "proj4",
-      attval = as.character(prj))
-  }
-
+  # The end --------------------------------------------------------------
 
-  #--- add variables values
-  for (k in seq_len(nl)) {
-    # write values to variable
-    vals <- raster::getValues(if (nl > 1) raster::raster(x, layer = k) else x)
-
-    try(ncdf4::ncvar_put(nc, varid = var_names[k],
-      vals = matrix(vals, ncol = var_chunksizes[2]),
-      start = var_start, count = var_chunksizes))
-
-    # add variable attributes
-    for (natt in ns_att_vars) {
-      ncdf4::ncatt_put(nc, varid = var_names[k], attname = natt,
-        attval = var_attributes[[natt]][k])
-    }
-
-    # Flush this step to the file so we dont lose it
-    # if there is a crash or other problem
-    ncdf4::nc_sync(nc)
-  }
+  message(paste("The file has", nc$nvars, "variables and",  nc$ndim,"dimensions"))
 
   invisible(TRUE)
 }
 
-#' @describeIn create_netCDF_from_raster_with_variables Convert array where
-#'   variables are organized in the third dimension to a \var{netCDF} file
+#' Populate an empty netcdf with values stored in an array
+#'
+#' This function add values to a structured netCDF created with the
+#' \code{\link[create_empty_netCDF_file]{create_empty_netCDF_file}} function.
+#' The data array provided as an argument to this function should be the same as
+#' that provided to the create_empty_netCDF_file function.
+#' One of \var{locations} or \var{grid} needs to be provided, but not both.
+#' If \var{locations} is a matrix or data.frame, \var{crs} needs to be provided.
+#'
+#' @param file A character string. File path to a pre-created, empty netCDF.
+#' @param data A numeric array.
+#' @param var_name A character string equal to the length of the number of variables.
```

Comment:
same as above, link to CMIP standard variable names

Reply:
fixed

Reply:
Not sure what this refers to.